firefox-ios/Client/Frontend/Settings/HomepageSettings/TopSitesSettings/TopSitesRowCountSettingsController.swift

```diff
@@ -33,6 +33,7 @@ class TopSitesRowCountSettingsController: SettingsTableViewController {
                 self.numberOfRows = num
                 self.prefs.setInt(Int32(num), forKey: PrefsKeys.NumberOfTopSiteRows)
                 self.tableView.reloadData()
+                NotificationCenter.default.post(name: .HomePanelPrefsChanged, object: nil)
```

Comment:
I'm a little bit concerned about this change, simply because this notification seems fairly descriptive about its purpose (home panel preferences changed) but this function `generateSettings()` is more of a multi-purpose utility func AFAICS. I see that we post this same notification in a few other spots, however. But are we using this here to gain some kind of side effect benefit? If so I'd rather see us rework this at some point, or even add a new notification if needed, rather than repurpose a notification that wasn't really intended for this.

Reply:
Yeah, I would tend to agree with this. This seems to not be so much tied to t fact that the panel changed, so this seems odd here.

Reply:
that notification is sent when the SettingsController disappears, which happens before the `TopSitesRowCountSettingsController` appears, so if you dismiss the screen by swiping down, that notification wouldn't have been sent anymore and it would lack a trigger to tell the Homepage that something changed, and `generateSettings()` is used to recalculate the layout and populate the rows with data, which is needed when the number of rows is modified.